### ISSUE TITLE
can't download ERROR: 403 Forbidden, retrying (attempt 1 )...

### ISSUE THREAD

**Issue by dexter101010** [Puyodead1/udemy-downloader#246]
[00:00:00] [udemy-downloader] [main:1843] INFO: > Fetching course information, this may take a minute...
[00:00:00] [udemy-downloader] [_get:1125] ERROR: Failed request https://www.udemy.com/course/zabbix-monitoring/learn
[00:00:00] [udemy-downloader] [_get:1126] ERROR: 403 Forbidden, retrying (attempt 0 )...
[00:00:00] [udemy-downloader] [_get:1125] ERROR: Failed request https://www.udemy.com/course/zabbix-monitoring/learn
[00:00:00] [udemy-downloader] [_get:1126] ERROR: 403 Forbidden, retrying (attempt 1 )...

I have the drm keys (keyfile.json) also the access_token (.env) 

Please advise!

**Comment by Puyodead1** [comment]
if its a course included in a subscription, use -sc and the --browser option, otherwise provide a valid bearer token

**Comment by dexter101010** [comment]
![image](https://github.com/user-attachments/assets/870c8e8d-a662-48a6-b009-1c0f32aa3d2e)

**Comment by Puyodead1** [comment]
.env is deprecated, and if you're still getting that error then you're probably getting a cloudflare captcha, and theres nothing that can be done about that

**Comment by swahpy** [comment]
> .env is deprecated, and if you're still getting that error then you're probably getting a cloudflare captcha, and theres nothing that can be done about that

hi @Puyodead1, although this issue was closed, I faced the issue again today and may I clarify that? The course is a bought course and the correct bearer token was added to .env file. It still showed 403 forbidden. So if .env was deprecated, then for such cases(downloading bought course), which bearer token is needed, how to download the course then? Thank you for any hints and information. Thank you.
